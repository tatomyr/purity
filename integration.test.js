const { createStore } = require('./__factory__.js')
const { htmx } = require('./__htmx__.js')

describe('factory & htmx', () => {
  let store, Component, Connected
  beforeEach(() => {
    defaultState = {
      something: 'something',
    }
    const stateHandler = (state = defaultState, action = {}) => {
      switch (action.type) {
        case 'INIT':
          return state
        case 'CHANGE_SOMETHING':
          return { ...state, something: action.something }
        default:
          return {}
      }
    }
    store = createStore(stateHandler)
    Connected = store.connect(
      ({ title, something }) => `
        <h1>${title}</h1>
        <p>${something}</p>
      `
    )
    Component = () => htmx({ Connected })`
      <div id="root"><Connected title=${'Hello!'} /></div>
    `
    document.body.innerHTML = '<div id="root"></div>'
    store.mount(Component)
  })
  it('should render connected component properly', () => {
    expect(Component()).toMatchSnapshot()
  })
  it('should reflect state changes', () => {
    store.dispatch({ type: 'CHANGE_SOMETHING', something: 'new thing' })
    expect(Component()).toMatchSnapshot()
  })
})
