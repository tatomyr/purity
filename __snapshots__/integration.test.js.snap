// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`factory & htmx should reflect state changes 1`] = `
"<div id=\\"root\\">
        <h1>Hello!</h1>
        <p>new thing</p>
      </div>"
`;

exports[`factory & htmx should render connected component properly 1`] = `
"<div id=\\"root\\">
        <h1>Hello!</h1>
        <p>something</p>
      </div>"
`;
